development/build/utils.js

```diff
@@ -218,8 +228,67 @@ function getPathInsideNodeModules(packageName, pathToFiles) {
   return targetPath;
 }
 
+/**
+ * Get the name for the current build.
+ *
+ * @param {object} options - The build options.
+ * @param {string} options.buildType - The build type of the current build.
+ * @param {boolean} options.applyLavaMoat - Flag if lavamoat was applied.
+ * @param {boolean} options.shouldIncludeSnow - Flag if snow should be included in the build name.
+ * @param options.environment
+ * @returns {string} The build name.
+ */
+function getBuildName({
+  environment,
+  buildType,
+  applyLavaMoat,
+  shouldIncludeSnow,
+}) {
+  if (environment === ENVIRONMENT.PRODUCTION) {
+    return 'MetaMask';
+  }
+
+  const mv3Str = process.env.ENABLE_MV3 ? ' MV3' : '';
+  const lavamoatStr = applyLavaMoat ? ' lavamoat' : '';
+  const snowStr = shouldIncludeSnow ? ' snow' : '';
+
+  return buildType === 'mmi'
+    ? `MetaMask Institutional ${mv3Str}`
+    : `MetaMask ${capitalize(buildType)}${mv3Str}${lavamoatStr}${snowStr}`;
+}
+
+/**
+ * Get the app ID for the current build. Should be valid reverse FQDN.
+ *
+ * @param {object} options - The build options.
+ * @param {string} options.buildType - The build type of the current build.
+ * @returns {string} The build app ID.
+ */
+function getBuildAppId({ buildType }) {
+  const baseDomain = 'io.metamask';
+  return buildType === 'main' ? baseDomain : `${baseDomain}.${buildType}`;
+}
+
+/**
+ * Get the image data uri for the svg icon for the current build.
+ *
+ * @param {object} options - The build options.
+ * @param {string} options.buildType - The build type of the current build.
+ * @returns {string} The image data uri for the icon.
+ */
+function getBuildIcon({ buildType }) {
+  const svgLogoPath =
+    BUILD_TYPES_TO_SVG_LOGO_PATH[buildType] ||
+    BUILD_TYPES_TO_SVG_LOGO_PATH.main;
+  const svg = readFileSync(svgLogoPath, 'utf8');
+  return `data:image/svg+xml,${encodeURIComponent(svg)}`;
```

Comment:
Is this approach too brittle? At least it'll throw if the file is missing so it should be caught in the CI pipeline at the build step if the file has moved